server/workers/shareScrollWorker.ts

```diff
@@ -0,0 +1,548 @@
+
+    const SAMPLE_FLAIR = 'saxifrage';
+    const SAMPLE_WEEKLY_CLICKS = 12345;
+    const SAMPLE_ALLTIME_CLICKS = 123456;
+    const { bannerBuffer, bannerWidth, bannerHeight } = await getBannerBase(
+      username,
+      dragonCount,
+      growingCount,
+      SAMPLE_WEEKLY_CLICKS,
+      SAMPLE_ALLTIME_CLICKS,
+      SAMPLE_FLAIR
+    );
+
+    // const { dragonStrip, stripWidth, stripHeight } =
+    //   await getDragonStrip(dragonIds);
+
+    // const frames = await createFrames(dragonStrip, stripWidth, stripHeight);
+
+    // const gif = await GIF.createGif({
+    //   delay: FRAME_DELAY,
+    //   repeat: 0,
+    //   format: 'rgb444',
+    // })
+    //   .addFrame(frames)
+    //   .toSharp();
+
+    // await gif.toFile(`${filePath}.tmp`);
+
+    const baseBanner = await sharp(bannerBuffer)
+      .gif()
+      .toFile(`${filePath}.tmp`);
+  } catch (error) {
+    console.error('Error in generateBannerToTemporary:', error);
+    await fs.unlink(`${filePath}.tmp`).catch(() => {});
+    throw error;
+  }
+}
+
+async function getBannerBase(
+  username: string,
+  dragonCount: number,
+  growingCount: number,
+  weeklyClicks: number,
+  allTimeClicks: number,
+  flair: string | null
+) {
+  try {
+    const startTime = performance.now();
+    console.log('Generating banner stats...');
+    const BANNER_WIDTH = 327;
+    const BANNER_HEIGHT = 61;
+    let compositeImage = createEmptyFrame(BANNER_WIDTH, BANNER_HEIGHT);
+    const composites: sharp.OverlayOptions[] = [];
+
+    // base
+    composites.push({
+      input: '/src/public/banner/base.webp',
+      top: 0,
+      left: 0,
+    });
+
+    // scrollname
+    const usernamePng = await textToPng(
+      username,
+      '16px Alkhemikal',
+      'fill: #dff6f5;'
+    );
+    const { height: usernameHeight, width: usernameWidth } =
+      await sharp(usernamePng).metadata();
+    composites.push({
+      input: usernamePng,
+      top: 9 + (16 - usernameHeight),
+      left: 118,
+    });
+
+    // flair
+    if (flair && flair !== '') {
+      const flairPath = `/src/public/items/${flair}.png`;
+      const flairImage = sharp(flairPath)
+        .greyscale()
+        .threshold(255)
+        .composite([
+          {
+            input: Buffer.from([255, 255, 255, Math.ceil(255 / 5)]),
+            raw: {
+              width: 1,
+              height: 1,
+              channels: 4,
+            },
+            tile: true,
+            blend: 'dest-in',
+          },
+          { input: flairPath, left: -1, top: -1 },
+        ])
+        .png();
+      const { height: flairHeight } = await flairImage.metadata();
+      composites.push({
+        input: await flairImage.toBuffer(),
+        left: 118 + usernameWidth + 4,
+        top: 16 - Math.floor((flairHeight ?? 0) / 2),
+      });
+    }
+
+    const statText = (statName: string, statNumber: number) => `
+      <tspan fill="#dff6f5">${statName}:</tspan> 
+      <tspan fill="#f2bd59">${statNumber.toLocaleString()}</tspan>
```

Comment:
Okay so I notice I did this wrong - it works on numbers, not strings, so I should have `parseInt`'d that first - but `toLocaleString` on numbers [allows for commas](https://stackoverflow.com/questions/2901102/how-to-format-a-number-with-commas-as-thousands-separators/17663871#17663871), i.e. `12,345` instead of `12345`.

If you don't care for commas though, that's alright.

Reply:
@endulum this is a fair point. `toLocaleString` uses the system's locale by default which can change, though unlikely in our case. pass in `en-gb` or something so it's certain.